container-chain-pallets/authorities-noting/src/lib.rs

```diff
@@ -0,0 +1,397 @@
+    parity_scale_codec::{Decode, Encode},
+    sp_inherents::{InherentIdentifier, IsFatalError},
+    sp_runtime::{traits::Hash as HashT, RuntimeString},
+    sp_std::prelude::*,
+};
+
+pub trait GetContainerChains {
+    fn container_chains() -> Vec<ParaId>;
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+    use parity_scale_codec::FullCodec;
+
+    use super::*;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        type SelfParaId: Get<ParaId>;
+
+        type RelayChainStateProvider: cumulus_pallet_parachain_system::RelaychainStateProvider;
+
+        type AuthorityId: sp_std::fmt::Debug + PartialEq + Clone + FullCodec + TypeInfo;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// The new value for a configuration parameter is invalid.
+        FailedReading,
+        FailedDecodingHeader,
+        NoAuthoritiesFound,
+    }
+
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
+            let mut weight = Weight::zero();
+
+            // We clear this storage item to make sure its always included
+            DidSetOrchestratorAuthorityData::<T>::kill();
+
+            weight += T::DbWeight::get().writes(1);
+
+            // The read onfinalizes
+            weight += T::DbWeight::get().reads(1);
+
+            weight
+        }
+
+        fn on_finalize(_: BlockNumberFor<T>) {
+            assert!(
+                <DidSetOrchestratorAuthorityData<T>>::exists(),
+                "Orchestrator chain authorities data needs to be present in every block!"
+            );
+        }
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn orchestrator_para_id)]
+    pub type OrchestratorParaId<T: Config> = StorageValue<_, ParaId, ValueQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        pub orchestrator_para_id: ParaId,
+        #[serde(skip)]
+        pub _config: sp_std::marker::PhantomData<T>,
+    }
+
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            GenesisConfig {
+                orchestrator_para_id: 1000u32.into(),
+                _config: Default::default(),
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+        fn build(&self) {
+            OrchestratorParaId::<T>::put(self.orchestrator_para_id);
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight((0, DispatchClass::Mandatory))]
```

Comment:
Just wondering, shouldn't we use a different weight hint here? To avoid the same issue as we have with `parachainSystem.setValdiationData` expressed [here](https://github.com/paritytech/polkadot-sdk/pull/2233).

Reply:
yep exactly!